qiskit/compiler/assemble.py

```diff
@@ -422,32 +422,67 @@ def _expand_parameters(circuits, run_config):
     parameter_binds = run_config.parameter_binds
     if parameter_binds or \
        any(circuit.parameters for circuit in circuits):
-
-        all_bind_parameters = [bind.keys()
-                               for bind in parameter_binds]
-        all_circuit_parameters = [circuit.parameters for circuit in circuits]
-
-        # Collect set of all unique parameters across all circuits and binds
-        unique_parameters = {param
-                             for param_list in all_bind_parameters + all_circuit_parameters
-                             for param in param_list}
-
-        # Check that all parameters are common to all circuits and binds
-        if not all_bind_parameters \
-           or not all_circuit_parameters \
-           or any(unique_parameters != bind_params for bind_params in all_bind_parameters) \
-           or any(unique_parameters != parameters for parameters in all_circuit_parameters):
+        param_mismatch = False
```

Comment:
I haven't looked at the code in depth, but as a first impression wouldn't it be easier to re-use the `_unroll_param_dict` method in the circuit? Since that method is used to expand the parameters in `assign_parameters({param_vec: [0, 1, ... ]}` it should also work here, right?

https://github.com/Qiskit/qiskit-terra/blob/e193de04d59f078f100176fbc63fbb6ca14e51a5/qiskit/circuit/quantumcircuit.py#L2039

Reply:
Yes, makes sense. Thanks a lot for the hint. I'll look into that because I'd like to come up with a simpler implementation.

Reply:
I've pushed an implementation based on `_unroll_param_dict ` a per your suggestion. Thanks a lot.